### ISSUE TITLE
We should revamp the "Send diagnostic"

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#4014]
Few things disturb me:
* Does not really route to either email or github
* We should probably better have a bug report
* Should be clear that this is English only if this is a bug report
* ...

**Comment by kelson42** [comment]
Updating this issue as we have already started this effort, also on Apple. Will be completed for 4.0.0.

**Comment by harshsomankar123-tech** [comment]
Hi @kelson42 @gouri-panda @MohitMaliFtechiz,

I've been working on a possible solution for this issue based on the recent Apple implementation, while also addressing a few Android-specific concerns.
Apple Issue : https://github.com/kiwix/kiwix-apple/issues/1313
PR: https://github.com/kiwix/kiwix-apple/pull/1330

My  approach :

* Moving the bug reporting flow out of `ErrorActivity` into a dedicated `BugReportActivity`, allowing users to submit reports without triggering app restarts or other crash-related side effects.

* Adding checkboxes that allow users to choose which diagnostic information they want to include (logs, ZIM list, locale, device information, file layout, etc.), giving them more control over privacy and shared data.

* Providing an English-only reporting screen to encourage reports that can be understood and discussed by the wider contributor community.

* Offering three reporting options:

  * **Email Report** (private, sent to `app-diagnostic@kiwix.org`)
  * **Open GitHub** (copies the selected diagnostics to the clipboard and opens the public issue tracker)
  * **Share** (uses Android's native Share Sheet, which is particularly useful for F-Droid users or devices without an email client configured)

* Following the Apple logic by hiding the reporting option for branded/custom apps via `isBrandedApp`.

I believe the GitHub option could also improve the visibility of user-reported issues for community contributors, making it easier to investigate, reproduce, and help resolve bugs.

Before I move further with the implementation, I'd appreciate your thoughts on the overall direction. If you have any suggestions, concerns, or refinements in mind, I'd be happy to incorporate them.

Thanks!

**Comment by harshsomankar123-tech** [comment]
Hi @kelson42 @MohitMaliFtechiz, I think this would be a good issue to work on next. I've mentioned both the Apple issue and the related PR for reference. PTAL and let me know your thoughts. Thanks!

**Comment by MohitMaliFtechiz** [comment]
> Adding checkboxes that allow users to choose which diagnostic information they want to include (logs, ZIM list, locale, device information, file layout, etc.), giving them more control over privacy and shared data.

@harshsomankar123-tech No, it should not. There is no benefit to giving this. Previously, this was the approach where the user could select what they wanted to send. But it is a bug/diagnostic report, so we need all the information. That's why all the details are needed.

> Moving the bug reporting flow out of ErrorActivity into a dedicated BugReportActivity, allowing users to submit reports without triggering app restarts or other crash-related side effects.

They are still doing the same. Both `ErrorActivity`, and `DiagnosticReportActivity` provide thier own implementation. App restart is required when there is a crash because MainActivity is already destroyed due to crash. 

> Email Report (private, sent to app-diagnostic@kiwix.org)

We already have this.

> Providing an English-only reporting screen to encourage reports that can be understood and discussed by the wider contributor community.

I guess you are talking about the bug report, not for the app UI. 

> Open GitHub (copies the selected diagnostics to the clipboard and opens the public issue tracker)

This option is good. But we should discuss this before implementing; there can be a bunch of tickets open for the diagnostic report as well. As we receive many diagnostic reports via email. But for bug/crash reports, we can add this option.

**Comment by soloturn** [comment]
> **AI-assisted research note.** Filed by agent driven by @soloturn via [GDD](https://siliconsaga.github.io/yggdrasil/gdd/).

@harshsomankar123-tech's "Open GitHub" option is close to how Terasology's crash reporter already does this, worth citing as concrete prior art: [MovingBlocks/CrashReporter](https://github.com/MovingBlocks/CrashReporter) (the shared Java Swing crash-report UI used by the Terasology engine and DestinationSol).

Its actual flow, traced through the source:

1. `PastebinUploadRunnable` ([source](https://github.com/MovingBlocks/CrashReporter/blob/master/cr-core/src/main/java/org/terasology/crashreporter/pages/PastebinUploadRunnable.java)) uploads the full crash content to Pastebin via `org.jpastebin`, with `PasteExpireDate.ONE_MONTH` and `pasteFormat("apache")` (their closest available format for a log dump).
2. `FinalActionsPanel` ([source](https://github.com/MovingBlocks/CrashReporter/blob/master/cr-core/src/main/java/org/terasology/crashreporter/pages/FinalActionsPanel.java)) then shows the resulting Pastebin link in a text field with a copy button.
3. A separate "Report Issue" button just opens a **bare** `github.com/MovingBlocks/Terasology/issues/new` in the system browser (`openInBrowser(properties.get(KEY.REPORT_ISSUE_LINK))`, no `?title=`/`?body=` prefill).
4. The user pastes the copied Pastebin link into the issue they compose themselves.

Two things worth carrying over, since it's a genuinely proven, years-old pattern (this project has done it since ~2014):

- It never posts to GitHub directly (no API call, no token) - it's "copy a link, then open the compose page for the human to submit." That sidesteps needing any GitHub credentials/scopes in the app itself, and keeps a human in the loop before anything public gets filed - relevant to @MohitMaliFtechiz's point about not wanting a flood of auto-filed diagnostic tickets.
- It doesn't put the raw diagnostic dump in the clipboard/issue body at all - only a short link to where the full content lives. For kiwix-android specifically (device info, ZIM list, file layout, logs could get large), pasting all of that directly into a GitHub issue body would be noisy and could hit practical size limits; a "upload the bundle somewhere linkable, copy just the link" step is worth considering over dumping raw text to the clipboard.

(Note: GitHub's own "New issue" URL does support `?title=`/`?body=` query-string prefill, which Terasology's implementation apparently doesn't use, so that part isn't something to copy verbatim.)
